ca/ca.go

```diff
@@ -179,25 +179,59 @@ func (ca *CAImpl) newIntermediateIssuer(root *issuer, intermediateKey crypto.Sig
 	if err != nil {
 		return nil, err
 	}
-	ca.log.Printf("Generated new intermediate issuer with serial %s and SKI %x\n", ic.ID, subjectKeyID)
+	ca.log.Printf("Generated new intermediate issuer %s with serial %s and SKI %x\n", ic.Cert.Subject, ic.ID, subjectKeyID)
 	return &issuer{
 		key:  intermediateKey,
 		cert: ic,
 	}, nil
 }
 
-func (ca *CAImpl) newChain(intermediateKey crypto.Signer, intermediateSubject pkix.Name, subjectKeyID []byte) *chain {
+// newChain generates a new issuance chain, including a root certificate and numIntermediates intermediates (at least 1).
+// The first intermediate will use intermediateKey, intermediateSubject and subjectKeyId.
+// Any intermediates between the first intermediate and the root will have their keys and subjects generated automatically.
+func (ca *CAImpl) newChain(intermediateKey crypto.Signer, intermediateSubject pkix.Name, subjectKeyID []byte, numIntermediates int) *chain {
+	if numIntermediates <= 0 {
+		panic("At least one intermediate must be present in the certificate chain")
+	}
 	root, err := ca.newRootIssuer()
 	if err != nil {
 		panic(fmt.Sprintf("Error creating new root issuer: %s", err.Error()))
 	}
-	intermediate, err := ca.newIntermediateIssuer(root, intermediateKey, intermediateSubject, subjectKeyID)
+
+	// The last N-1 intermediates build a path from the root to the leaf signing certificate.
+	// If numIntermediates is only 1, then no intermediates will be generated here.
+	prev := root
+	intermediates := make([]*issuer, numIntermediates)
+	for i := 0; i < numIntermediates-1; i++ {
+		k, ski, err := makeKey()
+		if err != nil {
+			panic(fmt.Sprintf("Error creating new intermediate issuer: %v", err))
+		}
+		intermediate, err := ca.newIntermediateIssuer(prev, k, pkix.Name{
+			CommonName: intermediateCAPrefix + hex.EncodeToString(makeSerial().Bytes()[:3]),
```

Comment:
Would probably be useful to specify what position in the chain this intermediate is, i.e. `fmt.Sprintf("%s %d", intermediateCAPrefix, i)` and skip the random suffix.

Reply:
I don't think that eliminating the random suffix would combine well with `PEBBLE_ALTERNATE_ROOTS`. We would end up with a "CN=Pebble Intermediate CA 2" in every chain, suggesting they're cross-signs, but they're actually totally unrelated with different subject keys.

What if we logged each issuer chain on startup, instead of logging the generation of each issuer? Helps the user while keeping opaque things opaque.

Reply:
How about having both the random suffix and the position? I.e. `Pebble Intermediate CA 3 2a9423`? Or even include the alternative root number, so we have `Pebble Root CA 2 3a74f` and `Pebble Intermediate CA 2:3 2a9423`?

Reply:
I'm not strongly opinionated either way.

Another option would be to generate a unique chain ID, so with two chains you'd have i.e. `root (2a9423) -> intermediate 1 (2a9423) -> intermediate 2 (2a9423) -> leaf` and `root (84799db) -> intermediate 1 (84799db) -> intermediate 2 (84799db) -> leaf`

Reply:
The leaf's issuer would need to have a suffix-less name ("Pebble Intermediate XYZ") as it must have the same subject across every chain.

I'll try experiment with a chain ID for all the issuers except that one, but otherwise, I'm probably just going to log the issuance chains and keep the names as-is.